content/en/docs/components/pipelines/v2/migration.md

```diff
@@ -0,0 +1,470 @@
+
+### Breaking changes
+
+#### create_component_from_func and func_to_container_op support
+
+**Affects:** KFP OSS users and Vertex AI Pipelines users
+
+`create_component_from_func` and `func_to_container_op` are both used in KFP SDK v1 to create lightweight Python function-based components.
+
+Both functions are removed in KFP SDK v2.
+
+**Change:** Use the [`@dsl.component`][dsl-component] decorator, as described in [Lightweight Python Components][lightweight-python-components] and [Containerized Python Components][containerized-python-components].
+
+<table>
+<tr>
+<th>Previous usage</th>
+<th>New usage</th>
+</tr>
+<tr>
+<td>
+
+```python
+from kfp.components import create_component_from_func
+from kfp.components import func_to_container_op
+
+@create_component_from_func
+def component1(...):
+    ...
+
+def component2(...):
+    ...
+
+component2 = create_component_from_func(component2)
+
+@func_to_container_op
+def component3(...):
+    ...
+
+@dsl.pipeline(name='my-pipeline')
+def pipeline():
+    component1(...)
+    component2(...)
+    component3(...)
+```
+
+</td>
+<td>
+
+```python
+from kfp import dsl
+
+@dsl.component
+def component1(...):
+    ...
+
+@dsl.component
+def component2(...):
+    ...
+
+@dsl.component
+def component3(...):
+    ...
+
+@dsl.pipeline(name='my-pipeline')
+def pipeline():
+    component1(...)
+    component2(...)
+    component3(...)
+```
+
+</td>
+</tr>
+</table>
+
+#### ContainerOp support
+
+**Affects:** KFP OSS users
+
+`ContainerOp` has been deprecated since mid-2020. `ContainerOp` instances do not carry a description of inputs and outputs and therefore cannot be compiled to [IR YAML][ir-yaml].
+
+`ContainerOp` is removed from v2.
+
+**Change:** Use the [`@dsl.container_component`][dsl-container-component] decorator as described in [Container Components][container-components].
+
+<table>
+<tr>
+<th>Previous usage</th>
+<th>New usage</th>
+</tr>
+<tr>
+<td>
+
+```python
+from kfp import dsl
+
+# v1 ContainerOp will not be supported.
+component_op = dsl.ContainerOp(...)
+
+# v1 ContainerOp from class will not be supported.
+class FlipCoinOp(dsl.ContainerOp):
+```
+
+</td>
+<td>
+
+```python
+from kfp import dsl
+
+@dsl.container_component
+def flip_coin(rand: int, result: dsl.OutputPath(str)):
+  return ContainerSpec(
+    image='gcr.io/flip-image'
+    command=['flip'],
+    arguments=['--seed', rand, '--result-file', result])
+```
+
+</td>
+</tr>
+</table>
+
+#### VolumeOp and ResourceOp support
+
+**Affects:** KFP OSS users
+
+`VolumeOp` and `ResourceOp` expose direct access to Kubernetes resources within a pipeline definition. There is no support for these features on a non-Kubernetes platforms.
+
+These features will be supported via platform-specific functionality, which includes Kubernetes-specific functionality. This functionality is a work in progress.
```

Comment:
This section definitely needs a rewrite when we go GA. Should we give more specifics on "work in progress" means ready by 2.0 GA?
Also the SDK syntax for this is ready, up to you to decide when to add the code snippet here.

Reply:
> This section definitely needs a rewrite when we go GA.

For GA we will have full platform-specific docs (both in KFP docs and in kfp-kubernetes library docs), with authoring examples to reference. Is there more you have in mind that I can capture in a TODO?

> Should we give more specifics on "work in progress" means ready by 2.0 GA?

Yes, will update.

> Also the SDK syntax for this is ready, up to you to decide when to add the code snippet here.

I prefer to wait as including may suggest that the BE supports it in the current state, as everything else in the docs is a reflection of the current state. Let me know if you disagree.